proto/info/edge_node_cluster.proto

```diff
@@ -0,0 +1,237 @@
+  uint32 pod_count = 2;
+
+  // Number of pods in the namespace that are running
+  uint32 pod_running_count = 3;
+
+  // Number of pods in the namespace that are pending
+  uint32 pod_pending_count = 4;
+
+  // Number of pods in the namespace that are failed
+  uint32 pod_failed_count = 5;
+
+  // Number of pods in the namespace that are succeeded
+  uint32 pod_succeeded_count = 6;
+}
+
+enum KubePodStatus {
+  KUBE_POD_STATUS_UNKNOWN = 0;
+  KUBE_POD_STATUS_PENDING = 1;
+  KUBE_POD_STATUS_RUNNING = 2;
+  KUBE_POD_STATUS_SUCCEEDED = 3;
+  KUBE_POD_STATUS_CONTAINER_CREATING = 4;
+  KUBE_POD_STATUS_CRASHLOOP_BACKOFF = 5;
+  KUBE_POD_STATUS_ERROR = 6;
+  KUBE_POD_STATUS_EVICTED = 7;
+  KUBE_POD_STATUS_FAILED = 8;
+}
+
+message KubeEVEAppPodInfo {
+  // Name of the EVE application
+  string name = 1;
+
+  // Application Status
+  KubePodStatus status = 2;
+
+  // Restart count of the application
+  uint32 restart_count = 3;
+
+  // Restart time of the application, seconds ago
+  google.protobuf.Timestamp restart_time = 4;
+
+  // Creation Time of the application
+  google.protobuf.Timestamp creation_timestamp = 5;
+
+  // IP address of the application, on cni0 interface
+  string ip_address = 6;
+
+  // Node name on which the application is running
+  string node_name = 7;
+}
+
+message KubeVolumeReplicaInfo {
+  // Name of the volume replica
+  string name = 1;
+
+  // Rebuild progress of the volume replica
+  uint32 rebuild_progress_percentage = 2;
+
+  // Replica status
+  StorageVolumeReplicaStatus status = 3;
+}
+
+message KubeVolumeInfo {
+  // Name of the volume
+  string name = 1;
+
+  // Status of the volume
+  StorageVolumeState state = 2;
+
+  // Robustness of the volume
+  StorageVolumeRobustness robustness = 3;
+
+  // Creation Time of the volume in the cluster
+  google.protobuf.Timestamp creation_timestamp = 4;
+
+  // Provisioned size of the volume in bytes
+  uint64 provisioned_bytes = 5;
+
+  // Allocated size of the volume in bytes
+  uint64 allocated_bytes = 6;
+
+  // PV/PVC status of the volume
+  StorageVolumePVCStatus pvc_status = 7;
+
+  // Replicas of the volume
+  repeated KubeVolumeReplicaInfo replica = 8;
+}
+
+message KubeStorageInfo {
+  // Overall status of Longhorn
+  StorageHealthStatus health = 1;
+
+  // time of the health status transition
+  google.protobuf.Timestamp transition_time = 2;
+
+  // Status of all the volumes in Longhorn
+  repeated KubeVolumeInfo volumes = 3;
+}
+
+enum KubeCompUpgradeStatus {
+  KUBE_COMP_UPGRADE_STATUS_UNKNOWN = 0;
+  KUBE_COMP_UPGRADE_STATUS_DOWNLOAD = 1;
+  KUBE_COMP_UPGRADE_STATUS_DOWNLOAD_FAILED = 2;
+  KUBE_COMP_UPGRADE_STATUS_IN_PROGRESS = 3;
+  KUBE_COMP_UPGRADE_STATUS_FAILED = 4;
+  KUBE_COMP_UPGRADE_STATUS_COMPLETED = 5;
+}
+
+enum KubeComp {
+  KUBE_COMP_UNKNOWN = 0;
+  KUBE_COMP_CONTAINERD = 1; //every node will publish
+  KUBE_COMP_K3S = 2; // every node will publish
+  KUBE_COMP_MULTUS = 3; // only the seed node will publish the remaining here and below
```

Comment:
Seed node might have died and been replaced months ago. So I think there might be an elected leader which will publish, and that leader would initially be the seed node, but the concept of a seed node is a transient designation for initial creation.

Reply:
Yes I'll fix this.  The first cluster node to upgrade was originally going to be the one listed as seed node, I'll change this to the 'first node to upgrade' to a given eve-os version.